src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
Use the destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ........... } = action;
  
    if (type === 'addProperties') {
```